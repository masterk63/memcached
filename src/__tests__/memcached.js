const Memcached = require('../models/Memcached');
const memcached = new Memcached().getInstance();

const mockData1 = {
  key: 'foo',
  value: 'say',
  exptime: 300,
  flags: 0
};

const mockData2 = {
  key: 'foo',
  value: 'kevin',
  exptime: 300,
  flags: 0
};

describe('Memcached logic', () => {
  test('Create key', () => {
    memcached.createKey(mockData1);
    expect(JSON.stringify(memcached.getInstance())).toMatchSnapshot();
    delete memcached.getInstance()[mockData1.key];
  });
  test('Read empty key', () => {
    const result = memcached.readKey(mockData1.key);
    expect(result).toBeUndefined();
  });
  test('Read key', () => {
    memcached.createKey(mockData1);
    const result = memcached.readKey(mockData1.key);
    expect(JSON.stringify(result)).toMatchSnapshot();
    delete memcached.getInstance()[mockData1.key];
  });
  test('UpdateKey key', () => {
    memcached.createKey(mockData1);
    memcached.updateKey(mockData2);
    expect(JSON.stringify(memcached.getInstance())).toMatchSnapshot();
    delete memcached.getInstance()[mockData1.key];
  });
  test('Delete key', () => {
    memcached.createKey(mockData1);
    memcached.deleteKeyCache(mockData1.key);
    expect(JSON.stringify(memcached.getInstance())).toMatchSnapshot();
  });
  test('Is Key Stored Key', () => {
    memcached.createKey(mockData1);
    const result = memcached.isKeyStored(mockData1.key);
    expect(result).toBe(true);
    delete memcached.getInstance()[mockData1.key];
  });
});
