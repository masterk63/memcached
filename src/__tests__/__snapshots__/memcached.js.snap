// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Memcached logic Create key 1`] = `"{\\"foo\\":{\\"key\\":\\"foo\\",\\"value\\":\\"say\\",\\"flags\\":0,\\"cas\\":1}}"`;

exports[`Memcached logic Delete key 1`] = `"{}"`;

exports[`Memcached logic Read key 1`] = `"{\\"key\\":\\"foo\\",\\"value\\":\\"say\\",\\"flags\\":0,\\"cas\\":2}"`;

exports[`Memcached logic UpdateKey key 1`] = `"{\\"foo\\":{\\"key\\":\\"foo\\",\\"value\\":\\"kevin\\",\\"exptime\\":300,\\"flags\\":0,\\"cas\\":4}}"`;
